tests/js/legendTests.js

```diff
@@ -0,0 +1,232 @@
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.registerNamespace("gpii.tests.chartAuthoring");
+
+    fluid.defaults("gpii.tests.chartAuthoring.pieChart.legend", {
+        gradeNames: ["gpii.chartAuthoring.pieChart.legend", "autoInit"],
+        legendOptions: {
+            colors: ["#000000", "#ff0000", "#00ff00", "#0000ff", "#aabbcc", "#ccbbaa"]
+        },
+        listeners: {
+            "onLegendCreated.addMouseoverListener": {
+                listener: "{that}.addD3Listeners",
+                args: ["{that}.dom.row", "mouseover", "gpii.tests.chartAuthoring.mouseOverListener"]
+            }
+        },
+        members: {
+            mouseOverListenerCalled: false
+        }
+    });
+
+    gpii.tests.chartAuthoring.objectArray = [{
+        id: "id0",
+        value: 15,
+        label: "One"
+    }, {
+        id: "id1",
+        value: 67,
+        label: "Two"
+    }, {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }, {
+        id: "id3",
+        value: 45,
+        label: "Four"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayAdd = [{
+        id: "id0",
+        value: 15,
+        label: "One"
+    }, {
+        id: "id1",
+        value: 67,
+        label: "Two"
+    }, {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }, {
+        id: "id3",
+        value: 45,
+        label: "Four"
+    },
+    {
+        id: "id4",
+        value: 26,
+        label: "Five"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayRemove = [{
+        id: "id0",
+        value: 15,
+        label: "One"
+    }, {
+        id: "id1",
+        value: 67,
+        label: "Two"
+    }, {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayChangeInPlace = [{
+        id: "id0",
+        value: 36,
+        label: "I"
+    }, {
+        id: "id1",
+        value: 67,
+        label: "II"
+    }, {
+        id: "id2",
+        value: 26,
+        label: "Three"
+    }];
+
+    gpii.tests.chartAuthoring.mouseOverListener = function (data, i, that) {
+        that.mouseOverListenerCalled = true;
+    };
+
+    // convenience function for easing testing of colors (jquery returns only RGB)
+    // Based off http://stackoverflow.com/questions/4262417/jquery-hex-to-rgb-calculation-different-between-browsers
+    gpii.tests.chartAuthoring.hexToRGB = function (hexStr){
+        // note: hexStr should be #rrggbb
+        var hex = parseInt(hexStr.substring(1), 16);
+        var r = (hex & 0xff0000) >> 16;
+        var g = (hex & 0x00ff00) >> 8;
+        var b = hex & 0x0000ff;
+        return "rgb(" + r + ", " + g + ", " + b + ")";
+    };
+
+    gpii.tests.chartAuthoring.testLegendSyncWithModelDataSet = function(that) {
+        var dataSet = that.model.dataSet;
+        //
+        if(that.options.legendOptions.sort) {
+            dataSet.sort(function(a,b) {
```

Comment:
Exactly same sorting function used in legend.js. If this piece of code is wrong, the test is not catching it. You probably wanna directly pass in the sorted array value for comparison.